stdlib/builtins.pyi

```diff
@@ -920,22 +920,28 @@ class range(Sequence[int]):
     def __repr__(self) -> str: ...
     def __reversed__(self) -> Iterator[int]: ...
 
-class property(object):
-    fget: Callable[[Any], Any] | None
-    fset: Callable[[Any, Any], None] | None
-    fdel: Callable[[Any], None] | None
+_F = TypeVar("_F", contravariant=True)
+_G = TypeVar("_G", covariant=True)
+
+class property(Generic[_F, _G]):
+    fget: Callable[[_F], _G] | None
+    fset: Callable[[_F, _G], None] | None
+    fdel: Callable[[_G], None] | None
     def __init__(
         self,
-        fget: Callable[[Any], Any] | None = ...,
-        fset: Callable[[Any, Any], None] | None = ...,
-        fdel: Callable[[Any], None] | None = ...,
+        fget: Callable[[_F], _G] | None = ...,
+        fset: Callable[[_F, _G], None] | None = ...,
+        fdel: Callable[[_G], None] | None = ...,
         doc: str | None = ...,
     ) -> None: ...
-    def getter(self, fget: Callable[[Any], Any]) -> property: ...
-    def setter(self, fset: Callable[[Any, Any], None]) -> property: ...
-    def deleter(self, fdel: Callable[[Any], None]) -> property: ...
-    def __get__(self, obj: Any, type: type | None = ...) -> Any: ...
-    def __set__(self, obj: Any, value: Any) -> None: ...
+    def getter(self, fget: Callable[[_F], _G]) -> property[_F, _G]: ...
+    def setter(self, fset: Callable[[_F, _G], None]) -> property[_F, _G]: ...
+    def deleter(self, fdel: Callable[[_G], None]) -> property[_F, _G]: ...
```

Comment:
Maybe the return type should be a subclass of `property` when called on an instance of the subclass? I haven't looked at mypy_primer errors yet, so I don't know whether this fixes them.

```suggestion
    def getter(self: _T, fget: Callable[[_F], _G]) -> _T: ...
    def setter(self: _T, fset: Callable[[_F, _G], None]) -> _T: ...
    def deleter(self: _T, fdel: Callable[[_G], None]) -> _T: ...
```

Reply:
Thanks, that sounds good. I think mypy still complains, but it seems like an improvement :)